packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -322,6 +325,26 @@ export interface CommonProjectProps {
      */
     buildSpec?: any;
 
+    /**
+     * Run a script from an asset as build script
+     *
+     * If supplied together with buildSpec, the asset script will be run
+     * _after_ the existing commands in buildspec.
+     *
+     * This feature can also be used without a source, to simply run an
+     * arbitrary script in a serverless way.
+     *
+     * @default No asset build script
+     */
+    buildScriptAsset?: assets.Asset;
```

Comment:
I guess we should have abstract `FileAsset` and `DirectoryAsset`, and this should be a `DirectoryAsset`, correct?

Reply:
Yes, that would be a good way to ensure that.